serial-settings/src/lib.rs

```diff
@@ -122,163 +116,377 @@ impl<'a, P: Platform<Y>, const Y: usize> Interface<'a, P, Y> {
     ) {
-        let key = menu::argument_finder(item, args, "item").unwrap().unwrap();
-        match settings.get_json(key, interface.buffer) {
-            Err(e) => {
-                writeln!(interface, "Failed to read {key}: {e}")
-            }
-            Ok(len) => {
-                writeln!(
-                    interface.platform.interface_mut(),
-                    "{key}: {}",
-                    core::str::from_utf8(&interface.buffer[..len]).unwrap()
-                )
-            }
-        }
-        .unwrap();
+        let key = menu::argument_finder(item, args, "path").unwrap();
+        Self::iter_root(
+            key,
+            interface,
+            settings,
+            |key, interface, settings, defaults| {
+                // Get default
+                let slic = postcard::ser_flavors::Slice::new(interface.buffer);
+                let slic = match defaults.get_postcard_by_key(key, slic) {
+                    Err(miniconf::Error::Traversal(Traversal::Absent(_))) => {
+                        log::warn!(
+                            "Can't clear. Default is absent: `{}`",
+                            key.as_str()
+                        );
+                        None
+                    }
+                    Err(e) => {
+                        writeln!(
+                            interface,
+                            "Failed to get default `{}`: {e}",
+                            key.as_str()
+                        )
+                        .unwrap();
+                        return;
+                    }
+                    Ok(slic) => Some(slic),
+                };
+
+                // Set default
+                if let Some(slic) = slic {
+                    let slic = postcard::de_flavors::Slice::new(slic);
+                    match settings.set_postcard_by_key(key, slic) {
+                        Err(miniconf::Error::Traversal(Traversal::Absent(
+                            _,
+                        ))) => {
+                            return;
+                        }
+                        Err(e) => {
+                            writeln!(
+                                interface,
+                                "Failed to set {}: {e:?}",
+                                key.as_str()
+                            )
+                            .unwrap();
+                            return;
+                        }
+                        Ok(_rest) => {
+                            interface.updated = true;
+                            writeln!(
+                                interface,
+                                "Cleared current `{}`",
+                                key.as_str()
+                            )
+                            .unwrap()
+                        }
+                    }
+                }
+
+                // Check for stored
+                match interface.platform.fetch(interface.buffer, key.as_bytes())
```

Comment:
It seems odd to me that `clear` isn't the inverse of `set`. `set` modifies in-memory settings, but doesn't touch the persistent settings (that's what `store` does). Is there something that's equivalent to the inverse of `set`? I.e. reset in-memory to default, but don't touch persistent settings?

Reply:
`clear` is the inverse of `set + store`.
we'd have to split `clear` into `default` (inverse to `set`) and `clear` (inverse to `store`).

Reply:
Yeah I guess the operator can just power cycle the unit to undo `set`